lib/content_publisher/featured_document_migrator.rb

```diff
@@ -0,0 +1,32 @@
+module ContentPublisher
+  class FeaturedDocumentMigrator
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      edition = document.published_edition.present? ? document.published_edition : document.latest_edition
+
+      public_url = Whitehall.url_maker.public_document_url(edition)
+      public_updated_at = (edition.public_timestamp || edition.updated_at)
+
+      features = Feature.includes(:feature_list).where(document_id: document.id)
```

Comment:
Might want to use `includes(feature_list: :featurable)` as we probably still have an n+1 going on for the parent.

Or we could switch back to parent_type and parent_id fields passed in, as they won't need a query.